packages/core/src/components/dialog/dialog.tsx

```diff
@@ -122,6 +126,13 @@ export class Dialog extends AbstractComponent<IDialogProps, {}> {
             </div>
         );
     }
+
+    private handleContainerClick = (evt: React.MouseEvent<HTMLDivElement>) => {
+        // quick re-implementation of canOutsideClickClose because .pt-dialog-container covers the backdrop
+        if (this.props.canOutsideClickClose && evt.currentTarget.closest(`.${Classes.DIALOG}`) == null) {
```

Comment:
currentTarget should be container. i'm testing that the click did not come from within the `.pt-dialog`, meaning it came from the `.pt-dialog-container` itself.